app/commands/track/update_build_status.rb

```diff
@@ -0,0 +1,359 @@
+class Track::UpdateBuildStatus
+  include Mandate
+
+  initialize_with :track
+
+  def call
+    track.build_status = build_status
+  end
+
+  private
+  def build_status
+    {
+      health:,
+      volunteers:,
+      students:,
+      submissions:,
+      mentor_discussions:,
+      syllabus:,
+      practice_exercises:,
+      test_runner:,
+      representer:,
+      analyzer:
+    }
+  end
+
+  def health
+    num_exemplar = component_health_statuses[:exemplar].to_i
+    num_healthy = component_health_statuses[:healthy].to_i
+
+    return :exemplar if num_exemplar == 5
+    return :healthy if num_exemplar + num_healthy == 5
+
+    :needs_attention
+  end
+
+  memoize
+  def component_health_statuses = [
+    analyzer_health,
+    test_runner_health,
+    representer_health,
+    syllabus_health,
+    practice_exercises_health
+  ].tally
+
+  def volunteers
+    volunteer_user_ids = User::ReputationPeriod.where(
+      period: :forever,
+      about: :track,
+      track_id: track.id
+    ).select(:user_id)
```

Comment:
I think you want to also set `category: :any` here. IIRC the index can only work if that is explicitely set.